test/mciproject.yml

```diff
@@ -58,69 +83,86 @@ tasks:
-                install_name_tool -change /usr/local/lib/libwiredtiger-$WT_VERSION.dylib $(pwd)/.libs/libwiredtiger-$WT_VERSION.dylib .libs/wt
-              fi
-
-              ${test_env_vars|} TESTUTIL_ENABLE_LONG_TESTS=1 ${make_command|make} VERBOSE=1 check 2>&1
-            fi
-      - command: archive.targz_pack
+
+            ${test_env_vars|} TESTUTIL_ENABLE_LONG_TESTS=1 ${make_command|make} VERBOSE=1 check ${smp_command|} 2>&1
+
+  - name: examples-c-test
+    depends_on:
+      - name: compile
+    commands:
+      - func: "fetch artifacts"
+      - func: "compile wiredtiger"
+      - command: shell.exec
         params:
-          target: "wiredtiger.tgz"
-          source_dir: "wiredtiger"
-          include:
-             - "./**"
-      - command: s3.put
+          working_dir: "wiredtiger"
+          script: |
+            set -o errexit
+            set -o verbose
+
+            ${test_env_vars|} ${make_command|make} VERBOSE=1 check -C examples/c  ${smp_command|} 2>&1
+
+  - name: csuite-col-append-test
+    depends_on:
+      - name: compile
+    commands:
+      - func: "fetch artifacts"
+      - func: "compile wiredtiger"
+      - command: shell.exec
         params:
-          aws_secret: ${aws_secret}
-          aws_key: ${aws_key}
-          local_file: wiredtiger.tgz
-          bucket: build_external
-          permissions: public-read
-          content_type: application/tar
-          display_name: Binaries
-          remote_file: wiredtiger/${build_variant}/${revision}/binaries/${build_id}.tgz
+          working_dir: "wiredtiger"
+          script: |
+            set -o errexit
+            set -o verbose
+
+            ${test_env_vars|} TESTUTIL_ENABLE_LONG_TESTS=1 $(pwd)/test/csuite/test_wt2246_col_append 2>&1
 
   - name: unit-test
     depends_on:
     - name: compile
     commands:
-      - func: "fetch binaries"
+      - func: "fetch artifacts"
       - command: shell.exec
         params:
           working_dir: "wiredtiger"
           script: |
             set -o errexit
             set -o verbose
 
-            # Avoid /usr/bin/python, at least on macOS: with System Integrity
-            # Protection enabled, it ignores DYLD_LIBRARY_PATH and hence
-            # doesn't find the WiredTiger library in the local tree.
-            ${test_env_vars|} python ./test/suite/run.py -v 2 ${smp_command|} 2>&1
+            if [ "Windows_NT" = "$OS" ]; then
+              ${test_env_vars|} python ./test/suite/run.py -v 2 ${smp_command|} 2>&1
+            elif [ "$(uname -s)" == "Darwin" ]; then
+              # Avoid /usr/bin/python, at least on macOS: with System Integrity
+              # Protection enabled, it ignores DYLD_LIBRARY_PATH and hence
+              # doesn't find the WiredTiger library in the local tree.
+              ${test_env_vars|} python ./test/suite/run.py -v 2 ${smp_command|} 2>&1
```

Comment:
Is this exactly the same as the Windows command?

Reply:
I tried to adjust the OS checking sequence as suggested, but got a failed [patch](https://evergreen.mongodb.com/version/5bf4ac0ee3c33109cf6bc2a1) - the `if [ "$(uname -s)" == "Linux" ]` checking did not seem working for some reason. I tried before on a ubuntu14.04-test spawn host before and `uname -s` returned `Linux`. 
Considering we are going to split unit-test task in WT-4410, in which the whole OS checking logic here will be simplified/removed, I'd prefer to keep the current change as is.